src/main/java/bleuauction/bleuauction_be/server/notice/service/NoticeComponentService.java

```diff
@@ -0,0 +1,97 @@
+package bleuauction.bleuauction_be.server.notice.service;
+
+import bleuauction.bleuauction_be.server.attach.entity.Attach;
+import bleuauction.bleuauction_be.server.attach.service.AttachService;
+import bleuauction.bleuauction_be.server.config.annotation.ComponentService;
+import bleuauction.bleuauction_be.server.member.entity.Member;
+import bleuauction.bleuauction_be.server.member.entity.MemberCategory;
+import bleuauction.bleuauction_be.server.ncp.NcpObjectStorageService;
+import bleuauction.bleuauction_be.server.notice.entity.Notice;
+import bleuauction.bleuauction_be.server.notice.repository.NoticeRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.util.List;
+
+@Slf4j
+@ComponentService
+@Transactional
+@RequiredArgsConstructor
+public class NoticeComponentService {
+
+    private final NoticeRepository noticeRepository;
+    private final AttachService attachService;
+    private final NcpObjectStorageService ncpObjectStorageService;
+
+
+    @Transactional
+    public Long enroll(Notice notice, List<MultipartFile> multipartFiles, Member member) {
+
+        if (!MemberCategory.A.equals(member.getMemberCategory())) {
+            throw new IllegalArgumentException("권한이 없습니다.");
+        }
```

Comment:
요놈이 아얘 중복되는데 이런건 어때여?
메소드로 이런걸 미리 만드는거에여
```java
private void isMemberAdmin(MemberCategory memberCategory) {
        if (!MemberCategory.A.equals(memberCategory)) {
            throw new IllegalArgumentException("권한이 없습니다.");
        }
}
``` 

이후 똑같은 코드를 아래처럼 바꾸는거죠
```suggestion
        isMemberAdmin(member.getMemberCategory());
```

이상 오지랖이었습니다.

Reply:
계속 반복되니까 하나 만들면 좋겠는데 라는 생각은 언젠가 한 번 스쳤는데,,, bbb!!!